blobstore-basics/src/main/java/org/jclouds/examples/blobstore/basics/MainApp.java

```diff
@@ -65,34 +67,41 @@
    public static final Set<String> allKeys = ImmutableSet.copyOf(Iterables.concat(appProviders.keySet(), allApis.keySet()));
    
    public static int PARAMETERS = 4;
-   public static String INVALID_SYNTAX = "Invalid number of parameters. Syntax is: \"provider\" \"identity\" \"credential\" \"containerName\" ";
+   public static String INVALID_SYNTAX = "Invalid number of parameters. Syntax is: \"provider\" \"identity\" \"credential\" \"containerName\" and optionally \"endpoint\".";
 
    public static void main(String[] args) throws IOException {
 
+      String provider;
+      String identity;
+      String credential;
+      String containerName;
+      String endpoint = null;
+      
       if (args.length < PARAMETERS)
          throw new IllegalArgumentException(INVALID_SYNTAX);
 
       // Args
-
-      String provider = args[0];
-
+      provider = args[0];
       // note that you can check if a provider is present ahead of time
       checkArgument(contains(allKeys, provider), "provider %s not in supported list: %s", provider, allKeys);
-
-      String identity = args[1];
-      String credential = args[2];
-      String containerName = args[3];
+      identity = args[1];
+      credential = args[2];
+      containerName = args[3];
+      if (args.length > 4) endpoint = args[4];
 
       // Init
-      BlobStoreContext context = ContextBuilder.newBuilder(provider)
-                                               .credentials(identity, credential)
-                                               .buildView(BlobStoreContext.class);
+      ContextBuilder contextBuilder = ContextBuilder.newBuilder(provider)
+              .credentials(identity, credential);
+      if (!Strings.isNullOrEmpty(endpoint)) {
+         contextBuilder.endpoint(endpoint);
+      }
+      BlobStoreContext context = contextBuilder.buildView(BlobStoreContext.class);
 
+      BlobStore blobStore = null;
       try {
-
          ApiMetadata apiMetadata = context.unwrap().getProviderMetadata().getApiMetadata();
+         blobStore = context.getBlobStore();
          // Create Container
-         BlobStore blobStore = context.getBlobStore();
          Location location = null;
```

Comment:
These changes disappeared in the latest PR, but you could allow `jclouds.region` property to set a non-default `location`.